services/provider/server/server.go

```diff
@@ -189,3 +206,142 @@ func newClient() (client.Client, error) {
+		})})
+
+	// Service for monitoring endpoints
+	scMonitoring := &v1.Service{}
+	err = s.client.Get(ctx, types.NamespacedName{Name: "rook-ceph-mgr", Namespace: s.namespace}, scMonitoring)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get rook-ceph-mgr service. %v", err)
+	}
+
+	if scMonitoring.Spec.ClusterIP == "" || strconv.Itoa(int(scMonitoring.Spec.Ports[0].Port)) == "" {
+		return nil, fmt.Errorf("service rook-ceph-mgr clusterIP or port is empty")
+	}
+
+	extR = append(extR, &pb.ExternalResource{
+		Name: "monitoring-endpoint",
+		Kind: "CephCluster",
+		Data: mustMarshal(map[string]string{
+			"MonitoringEndpoint": scMonitoring.Spec.ClusterIP,
+			"MonitoringPort":     strconv.Itoa(int(scMonitoring.Spec.Ports[0].Port)),
+		})})
+
+	for _, i := range consumerResource.Status.CephResources {
+		switch i.Kind {
+		case "CephClient":
+			clientSecretName, err := s.getCephClientSecretName(ctx, i.Name)
+			if err != nil {
+				return nil, err
+			}
+
+			cephUserSecret := &v1.Secret{}
+			err = s.client.Get(ctx, types.NamespacedName{Name: clientSecretName, Namespace: s.namespace}, cephUserSecret)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get %s secret. %v", clientSecretName, err)
+			}
+
+			idProp := "userID"
+			keyProp := "userKey"
+			if strings.Contains(i.Name, "-cephfs-") {
+				idProp = "adminID"
+				keyProp = "adminKey"
+			}
+			extR = append(extR, &pb.ExternalResource{
+				Name: clientSecretName,
+				Kind: "Secret",
+				Data: mustMarshal(map[string]string{
+					idProp:  i.Name,
+					keyProp: string(cephUserSecret.Data[i.Name]),
+				}),
+			})
+		case "CephBlockPool":
+			nodeCephClientSecret, err := s.getCephClientSecretName(ctx, i.CephClients["node"])
+			if err != nil {
+				return nil, err
+			}
+
+			provisionerCephClientSecret, err := s.getCephClientSecretName(ctx, i.CephClients["provisioner"])
+			if err != nil {
+				return nil, err
+			}
+
+			extR = append(extR, &pb.ExternalResource{
+				Name: "ceph-rbd",
+				Kind: "StorageClass",
+				Data: mustMarshal(map[string]string{
+					"clusterID":                 s.namespace,
+					"pool":                      i.Name,
+					"imageFeatures":             "layering",
+					"csi.storage.k8s.io/fstype": "ext4",
+					"imageFormat":               "2",
+					"csi.storage.k8s.io/provisioner-secret-name":       provisionerCephClientSecret,
+					"csi.storage.k8s.io/node-stage-secret-name":        nodeCephClientSecret,
+					"csi.storage.k8s.io/controller-expand-secret-name": provisionerCephClientSecret,
+				})})
+		}
+	}
+
+	return extR, nil
+}
+
+func (s *OCSProviderServer) getCephClientSecretName(ctx context.Context, name string) (string, error) {
+	cephClient := &rookCephv1.CephClient{}
+	err := s.client.Get(ctx, types.NamespacedName{Name: name, Namespace: s.namespace}, cephClient)
+	if err != nil {
+		return "", fmt.Errorf("failed to get cephClient %s secret. %v", name, err)
+	}
+	return cephClient.Status.Info["secretName"], nil
```

Comment:
`cephClient` might not have a status when you try to read it (we might read it before it is reconcile by rook)
You need to protect against this access or else it will result in a panic.

Same goes for the Info map if the Info map is optional on the Status structure (not sure about that, you will need to check the type definition)